Library/Formula/node.rb

```diff
@@ -73,6 +73,10 @@ def install
       cd buildpath/"npm_install" do
         system "./configure", "--prefix=#{libexec}/npm"
         system "make", "install"
+        # `package.json` has relative paths to the npm_install directory.
+        # This copies back over the vanilla `package.json` that is expected.
+        # https://github.com/Homebrew/homebrew/issues/46131#issuecomment-157845008
+        cp buildpath/"npm_install/package.json", libexec/"npm/lib/node_modules/npm"
```

Comment:
If new things are introduced to the file, this would copy `package.json` from the tarball to the installation location.  Maybe I am missing something, but I think `cp`'ing the newer `package.json` addresses the concern.